### PR TITLE
Updating test matrix

### PR DESCRIPTION
**Pull request by lillian542**
Adds a table after the primary plugin matrix with the results of the IBMQ tests for Qiskit, run twice weekly. 

Also adds a cell in the `rc` plugin matrix that links to the new `catalyst-rc` test action on the catalyst repo.

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-49455]

File: .github/workflows/catalyst-latest-rc.yml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    - cron: '35 10 * * *'
+  workflow_dispatch:
+
+
+env:
+  PLUGIN_REPO: PennyLaneAI/catalyst
+  PLUGIN_BRANCH: master
+  PLUGIN_PACKAGE: catalyst
+  PENNYLANE_BRANCH: master
+  GCC_VERSION: 11   # What is this??
```
Comment (timmysilv) [inline .github/workflows/catalyst-latest-rc.yml:34]: lightning [needs it](https://github.com/PennyLaneAI/plugin-test-matrix/blob/e448c3491813dff2a3e279caa3a287dcd1ec6e4e/.github/workflows/lightning-latest-rc.yml#L42-L44) to install the right version of things. that said, lightning workflows were removed from this repo in #50, so that file probably shouldn't even exist. this seems to use whatever GCC is installed with the runner, which might not be great. idk

File: README.md [hunk 2]
```diff
@@ -34,6 +34,15 @@ All entries in the matrix are tested against PennyLane latest (GitHub master).
 
 * The Braket plugin device integration tests are run with `-k “not Sample and not no_0_shots”`,
   see #6
+  
+* The standard Qiskit plugin tests are run with `-k 'test_ibmq.py or test_runtime.py'`.
```
Comment (timmysilv) [inline README.md:38]:
```suggestion
* The Qiskit plugin integration tests are run with `-k 'test_ibmq.py or test_runtime.py'`. 
```

File: .github/workflows/catalyst-latest-rc.yml (hunk 1 above)
Comment (lillian542) [reply .github/workflows/catalyst-latest-rc.yml:34]: I see. Agreed with @dime10 that we will have this action [here](https://github.com/PennyLaneAI/catalyst/pull/406) in Catalyst instead, once that is merged I'll remove this and add a badge in the RC matrix referencing that. That version doesn't have this, so I guess we should be good on that front.

File: .github/workflows/catalyst-latest-rc.yml
Comment (mudit2812) [inline .github/workflows/catalyst-latest-rc.yml:1]: Now that you're directly referencing the latest-rc workflow from the catalyst repo in the README, is this file needed?

Comment (lillian542) [reply .github/workflows/catalyst-latest-rc.yml:1]: No, thanks, that was a failed attempt to make this work. Will remove.

**Review by mudit2812 — APPROVED** [review]
Looks good to me. The README is rendering properly. I guess we'll have to wait until feature freeze to see if the catalyst latest-rc row is working correctly.
